patches/tModLoader/Terraria/GameContent/UI/States/UIAchievementsMenu.cs.patch

```diff
@@ -0,0 +1,224 @@
+--- src/TerrariaNetCore/Terraria/GameContent/UI/States/UIAchievementsMenu.cs
++++ src/tModLoader/Terraria/GameContent/UI/States/UIAchievementsMenu.cs
+@@ -1,13 +_,16 @@
++using System;
+ using System.Collections.Generic;
+ using Microsoft.Xna.Framework;
+ using Microsoft.Xna.Framework.Graphics;
+ using ReLogic.Content;
++using ReLogic.OS.Windows;
+ using Terraria.Achievements;
+ using Terraria.Audio;
+ using Terraria.GameContent.UI.Elements;
+ using Terraria.GameInput;
+ using Terraria.ID;
+ using Terraria.Localization;
++using Terraria.ModLoader.UI;
+ using Terraria.UI;
+ using Terraria.UI.Gamepad;
+ 
+@@ -20,9 +_,22 @@
+ 	private List<UIToggleImage> _categoryButtons = new List<UIToggleImage>();
+ 	private UIElement _backpanel;
+ 	private UIElement _outerContainer;
++	private UIPanel blockingPanel;
++
++	private UIInputTextField uISearchBar;
++	private UIPanel uISearchBackground;
++
++	private bool _filterModded = false;
++
++	Asset<Texture2D> blockingBackground;
++	Asset<Texture2D> blockingBorder;
++
++	private UIPanel achievementResetAreYouSure;
+ 
+ 	public void InitializePage()
+ 	{
++		blockingBackground = ModLoader.ModLoader.ManifestAssets.Request<Texture2D>($"Terraria.ModLoader.UI.InnerPanelNotRounded");
++		blockingBorder = ModLoader.ModLoader.ManifestAssets.Request<Texture2D>($"Terraria.ModLoader.UI.PanelBorderNotRounded");
+ 		RemoveAllChildren();
+ 		_categoryButtons.Clear();
+ 		_achievementElements.Clear();
+@@ -50,22 +_,36 @@
+ 		_achievementsList.Top.Set(50f, 0f);
+ 		_achievementsList.ListPadding = 5f;
+ 		uIPanel.Append(_achievementsList);
++
+ 		UITextPanel<LocalizedText> uITextPanel = new UITextPanel<LocalizedText>(Language.GetText("UI.Achievements"), 1f, large: true);
+ 		uITextPanel.HAlign = 0.5f;
+ 		uITextPanel.Top.Set(-33f, 0f);
+ 		uITextPanel.SetPadding(13f);
+ 		uITextPanel.BackgroundColor = new Color(73, 94, 171);
+ 		uIElement.Append(uITextPanel);
++
+ 		UITextPanel<LocalizedText> uITextPanel2 = new UITextPanel<LocalizedText>(Language.GetText("UI.Back"), 0.7f, large: true);
+-		uITextPanel2.Width.Set(-10f, 0.5f);
++		uITextPanel2.Width.Set(-10f, 0.8f);
+ 		uITextPanel2.Height.Set(50f, 0f);
+ 		uITextPanel2.VAlign = 1f;
+-		uITextPanel2.HAlign = 0.5f;
++		uITextPanel2.HAlign = 0f;
```

Comment:
Perhaps instead of editing the values here, you can override them down where you make the `resetAchievementsButton`